### PR TITLE
e2e: extend FUSE heavy gates and SQLite WAL baseline

### PR DESCRIPTION
**Pull request by qiffang**
## Summary

Follow-up to merged PR #508. This PR carries the post-merge baseline delta from `dat9-dev2/fuse-performance-baseline` to `main` so the stacked compare PR #510 has a real merged base.

It includes:
- SQLite WAL performance baseline metrics alongside rollback-journal metrics.
- Daily/manual heavy FUSE gate wiring with Drive9 metrics archival.
- JuiceFS-style opt-in POSIX/fsx gate.
- FUSE concurrency stress artifact preservation and final correctness fixes for live mount `final/...` empty-read regressions.
- Regression coverage for FUSE rename/Fsync live inode/cache refresh paths.

## Validation

Local:
- Focused FUSE regressions.
- `go test ./pkg/fuse -count=1 -timeout=5m`
- `go test ./cmd/drive9 ./cmd/drive9/cli ./pkg/fuse -count=1 -timeout=5m`
- `go vet ./pkg/fuse ./cmd/drive9 ./cmd/drive9/cli`
- Focused `go test -race ./pkg/fuse ...`
- `PATH="/usr/local/go/bin:$PATH" make lint`
- `git diff --check`

GitHub Actions:
- All-flags local-e2e run: https://github.com/mem9-ai/drive9/actions/runs/27115563321
- Passed: release gate, performance artifact upload, Drive9 archive, FUSE concurrency stress, POSIX/fsx, final `Enforce smoke results`.
- Concurrency artifact: mounted manifest and remote manifest match expected; reader-errors is empty.

## Stacking note

PR #510 remains stacked on this branch and should be retargeted/rebased to `main` after this PR merges.

## Summary by CodeRabbit

* **New Features**
  * Opt-in POSIX/fsx FUSE validation gate with artifact uploads, workflow input, and job-summary visibility.
  * Performance baseline extended with WAL checkpoint metrics and a metrics comparison/archival toolchain (CLI + warning reporter).

* **Bug Fixes**
  * Richer per-file verification/mismatch reporting.
  * Improved snapshot/rename/unlink/fsync handling to preserve open-handle correctness and inode size/metadata consistency.

* **Tests**
  * Many new regression/unit tests for snapshots, renames, truncation, writebuffer, fsync/multipart, direct-IO, and concurrency.

* **Documentation**
  * Expanded E2E and agent docs covering knobs, inputs, and performance-compare guidance.

### CODE REVIEW

**Comment by qiffang** [comment]
R1 verdict: REVISE.

Blocking issue: fresh validation for the rebased PR #513 head `0edd8c9` is red. The required all-flags workflow run <https://github.com/mem9-ai/drive9/actions/runs/27105157529> fails in `Run FUSE release gate`, and final `Enforce smoke results` fails because that required gate is `failure`.

Concrete failure from the run log:

- SQLite rollback/WAL/churn start and initial concurrency writer loop reports `committed 40/40`.
- The same workload then reopens/fingerprints the DB and fails with `AssertionError: event_count=0 want=40`.
- Result line is `RESULT: 12/13 passed, 1 failed` for the SQLite correctness workload.

This invalidates the current PR claim that the rebased baseline follow-up is merge-ready. The older `0232ab2` green run cannot cover this PR because the branch was rebased to `main` and the fresh head changed to `0edd8c9`.

Also note: the stacked PR #510 fresh compare run <https://github.com/mem9-ai/drive9/actions/runs/27105182595> is red too, because it inherits the same release-gate failure before compare/archive can be trusted.

Required before next review:

1. Root-cause and fix the SQLite concurrency `event_count=0` regression on the rebased baseline branch.
2. Push a new PR #513 head and rerun the all-flags gate with SQLite WAL/churn/concurrency, performance baseline, Drive9 archive, FUSE concurrency stress, and POSIX/fsx enabled.
3. Rebase PR #510 onto the fixed PR #513 head and rerun the compare/archive gate.
4. Post fresh run/artifact evidence for the new heads; do not cite old `0232ab2` / `2c88b04` runs as final merge evidence.

**Comment by qiffang** [comment]
Updated baseline head to `40e4a54` and reran the fresh all-flags gate.

Fix summary:
- Clean sibling writable handles are now rebound to a lazy remote-backed buffer when another handle advances the committed revision for the same path.
- This closes the SQLite concurrency failure where final reopen could preload stale clean bytes after the writer had committed/checkpointed rows.
- Added regression `TestOpenWritablePreloadSkipsCleanSiblingReboundToNewRevision`.

Local validation:
- focused FUSE regressions
- `go test ./pkg/fuse -count=1 -timeout=5m`
- `go test ./cmd/drive9 ./cmd/drive9/cli ./pkg/fuse -count=1 -timeout=5m`
- `go vet ./pkg/fuse ./cmd/drive9 ./cmd/drive9/cli`
- `PATH="/usr/local/go/bin:$PATH" make lint`
- focused `go test -race ./pkg/fuse ...`
- `git diff --check`

Fresh all-flags validation:
- Run: https://github.com/mem9-ai/drive9/actions/runs/27105644847
- Head: `40e4a54bbe0380365bffacb2ef7f56c3265d511f`
- Result: success
- Steps verified green: release gate, performance artifact upload, Drive9 archive, FUSE concurrency stress, concurrency artifact upload, POSIX/fsx gate, final enforce.

Artifact checks:
- Concurrency `actual-mounted-manifest.json` == `expected-manifest.json`
- Concurrency `actual-remote-manifest.json` == `expected-manifest.json`
- `reader-errors.log` is empty
- SQLite rollback and WAL metrics both include `payload_verified_rows=256` and `integrity_check=ok`; WAL checkpoint has `checkpoint_busy=0`.
- Drive9 archive path: `/benchmarks/fuse-performance/2026/06/07/dat9-dev2-fuse-performance-baseline/40e4a54bbe03/27105644847-1`

**Comment by qiffang** [comment]
R2 verdict: GREEN for PR #513 at head `40e4a54`.

Fresh evidence after the rebased fix:

- PR #513 all-flags run <https://github.com/mem9-ai/drive9/actions/runs/27105644847> is green at head `40e4a54`.
- Required steps are all success: `Run FUSE release gate`, performance artifact upload, Drive9 archive, `Run FUSE concurrency stress`, `Run FUSE POSIX/fsx gate`, and final `Enforce smoke results`.
- Downloaded `fuse-concurrency-stress` artifact: mounted manifest and remote manifest both exactly match expected (`72` entries), and `reader-errors.log` is empty.
- Downloaded performance metrics artifact: rollback and WAL SQLite read metrics have `payload_verified_rows=256` and `integrity_check=ok`; WAL checkpoint has `checkpoint_busy=0`.
- PR checks are green (`ci`, `drive9-server-local e2e smoke`, CodeRabbit), and PR is `MERGEABLE`.

The previous blocker (`event_count=0 want=40` on rebased head `0edd8c9`) is closed by this fresh run. No remaining blocker from my side for PR #513.

Separate stacked note: PR #510 fresh compare run <https://github.com/mem9-ai/drive9/actions/runs/27105662353> is also workflow-green at head `5e4a9fa`; compare JSON produced `19` comparisons including rollback + WAL metrics. Its compare report status is `warning` because warning-only performance regressions were detected against the branch baseline, not because shape/validation failed.

**Review by mornyx — CHANGES_REQUESTED** [review]
### Review verdict: request changes

I re-reviewed the latest head `4f47e78` because the PR moved after the previous green comment on `40e4a54`. The new performance-compare tooling looks structurally sound to me, and the Python compare unit tests pass locally with `python3 -m unittest scripts/compare_fuse_performance_metrics_test.py`. I still see two correctness blockers in the FUSE handle refresh path:

1. `pkg/fuse/dat9fs.go:839`: `refreshCommittedRevisionForOpenHandles` skips any sibling whose mutex is held. That avoids deadlock, but a locked sibling can simply be doing a read or other non-commit operation, so it may never run a commit path that refreshes itself. If that sibling has a clean preloaded writable buffer, later reads can still hit the `dirty-clean-buffer` path and return bytes from the old revision. Please add a deferred refresh/adopt path for skipped clean siblings, or make the read path re-adopt and rebind clean buffers before serving loaded clean writable-buffer ranges.

2. `pkg/fuse/dat9fs.go:913`: `rebindCleanWriteBufferToRemoteLocked` resets `fh.Dirty` to the committed size but leaves `fh.OrigSize` at the size from the original open. After another handle commits a truncate or shorter replacement, `unlinkSnapshotSizeLocked` takes the max of `OrigSize`, inode size, and dirty size, so a later unlink/replace-open snapshot can attempt to read the old larger size from the now-shorter remote object and fail with `io.ErrUnexpectedEOF`. Please update `OrigSize` to the committed size when rebinding/finalizing committed state; the writer handle commit/finalize paths should also keep `OrigSize` aligned with the newly committed size.

Until these are addressed I would not approve this PR.

**Comment by qiffang** [comment]
Addressed latest review comments in `b1b554c`.

Changes:
- Locked clean siblings: added read-time clean-handle refresh/rebind before serving loaded clean writable-buffer bytes. This closes the case where `refreshCommittedRevisionForOpenHandles` skips a locked sibling that is not on a later commit path.
- `OrigSize` synchronization: `rebindCleanWriteBufferToRemoteLocked`, writer commit/finalize paths, and debounced flush success now align `fh.OrigSize` with committed handle size so unlink/replace-open snapshots do not read stale larger ranges.
- Snapshot remote fallback: `readOpenHandleSnapshot` now uses the shared read-slot-limited retry path instead of a single raw `doRangeRead` attempt.
- SQLite `-shm` local overlay: `*-shm` is now fail-closed unless an owning DB/WAL/journal handle is open, so unrelated `notes-shm` style paths no longer route to transient overlay.
- Tests updated for pending-index seeding and locked-helper use; added regressions for skipped clean-handle refresh, retrying snapshot reads, and confirmed SQLite sidecar routing.

Validation:
- `go test ./pkg/fuse -count=1 -timeout=5m`
- `go test ./cmd/drive9 ./cmd/drive9/cli ./pkg/fuse -count=1 -timeout=5m`
- `python3 -m unittest scripts/compare_fuse_performance_metrics_test.py`
- `go vet ./pkg/fuse ./cmd/drive9 ./cmd/drive9/cli`
- `PATH="/usr/local/go/bin:$PATH" make lint`
- focused `go test -race ./pkg/fuse ...`
- `git diff --check`

File: pkg/fuse/dat9fs_test.go [hunk 1]
```diff
@@ -11559,18 +11936,503 @@ func TestRenamePendingNewCommitUsesRemoteRenameWhenCanceledUploadBecameVisible(t
...
-		case r.Method == http.MethodPut && r.URL.Path == "/v1/fs"+newP:
-			finalPuts.Add(1)
-			body, err := io.ReadAll(r.Body)
-			if err != nil {
+		case r.Method == http.MethodGet && r.URL.Path == "/v1/status":
+			_ = json.NewEncoder(w).Encode(map[string]any{
+				"status": "active",
+				"limits": map[string]any{"inline_threshold": 1},
+			})
+			return
```
[the remark this answers is not in the document]
Comment (qiffang) [reply pkg/fuse/dat9fs_test.go:12329]: Current head already makes the multipart precondition explicit: this test calls `c.SetSmallFileThresholdForTests(1)` before constructing `Dat9FS` and exercising the fsync/release/rename path, so the body is forced through multipart regardless of defaults. The `/v1/status` handler also returns `inline_threshold=1`; the test no longer depends on incidental payload size vs default threshold.

File: pkg/fuse/dat9fs_test.go [hunk 2]
```diff
@@ -289,6 +291,130 @@ func TestUnknownInlineThresholdNonEmptyHandleUploadUsesMultipart(t *testing.T) {
...
+	fs.inodes.UpdateRevision(ino, baseRev)
+	fh := &FileHandle{
+		Ino:      ino,
+		Path:     filePath,
+		Dirty:    fs.newWriteBuffer(filePath, maxPreloadSize, 0),
+		OrigSize: origSize,
+		BaseRev:  baseRev,
+	}
+	fs.rebindCleanWriteBufferToRemoteLocked(fh, origSize)
+
```
[the remark this answers is not in the document]
Comment (qiffang) [reply pkg/fuse/dat9fs_test.go:393]: Fixed in `b1b554c`: the test now wraps the `rebindCleanWriteBufferToRemoteLocked` call with `fh.Lock()` / `fh.Unlock()`. I also reran the focused race coverage for this path successfully.

**Comment by qiffang** [comment]
Fresh all-flags validation on the fixed head is green:

- Head: `b1b554c`
- Run: https://github.com/mem9-ai/drive9/actions/runs/27115563321
- Passed: release gate, performance artifact upload, Drive9 archive, FUSE concurrency stress, POSIX/fsx, and final `Enforce smoke results`.
- Downloaded artifacts checked: concurrency mounted manifest == expected, remote manifest == expected, `reader-errors.log` is empty.
- SQLite metrics checked: rollback and WAL read workloads both have `payload_verified_rows=256` and `integrity_check=ok`; WAL checkpoint has `checkpoint_busy=0`.

**Review by srstack — APPROVED** [review]
## Review: PR #513 (updated)

**Status: Approved.** CI green, all checks pass.

I reviewed the full diff including the latest commit `b1b554c4` ("fix: refresh clean fuse handles after commits"). That commit resolves the main concerns I had from the earlier version.

### What the latest commit fixes well

- **`fh.OrigSize = committedSize`** in `rebindCleanWriteBufferToRemoteLocked`, `markHandleRemoteCommittedLocked`, and `finalizeHandleFlushLocked` — now `unlinkSnapshotSizeLocked` sees accurate committed sizes.
- **`readOpenHandleSnapshot` now uses `acquireRemoteReadSlot` + `readStreamRangeWithRetry`** instead of bare `doRangeRead` — transient failures are retried and the shared read-slot limiter is honored.
- **`refreshCleanCommittedRevisionForHandleLocked` called in `Read`** — when a clean writable handle had its commit skipped (TryLock), the next Read now detects the stale revision and rebinds to remote data before serving bytes.
- **`clearReadTargetForLockedHandle(fh)` in `rebindCleanWriteBufferToRemoteLocked`** — prevents reads from going to stale S3 presigned URLs after rebind.
- **`isConfirmedSQLiteWALIndexPath`** — transient overlay for `-shm` paths now requires a confirmed open db-sidecar handle, preventing false positives.
- **`canPatchExisting` guard** (`size <= fh.OrigSize`) in multipart patch path — prevents oversized patches.
- **Test `TestOpenReadOnlyPendingShadowUsesDirectIO`** now seeds PendingIndex.
- **Test `TestReadCleanWritableHandleRefreshesSkippedCommittedRevision`** — covers the lazy-refresh Read path.
- **Test `TestReadOpenHandleSnapshotRetriesTransientRangeRead`** — verifies retry on transient failures.

### Remaining observations (non-blocking)

1. **`rebindCleanWriteBufferToRemoteLocked` (`dat9fs.go:~966`)** creates `next := fs.newWriteBuffer(...)` and sets `fh.Dirty = next`, discarding the old buffer (`old := fh.Dirty`). If `old` had loaded part data with backing file descriptors, those are not explicitly closed. However, the guard `fh.Dirty.HasDirtyParts()` prevents rebinding when dirty, so this is low-risk.

2. **`ShadowStore.Rename` (`shadow.go:~584`)** now holds the lock during `os.Rename`. This is a correctness improvement over the old in-memory-first-then-disk approach, but note there is a small window: if the process crashes after `os.Rename` succeeds but before the in-memory maps are updated, the shadow file is at the new path but in-memory still points to the old path. Recovery would require a shadow-store scan on restart. Acceptable tradeoff vs. the old bug (in-memory pointing to nonexistent shadow on disk failure).

3. **CI workflow `local-e2e.yml` line ~60** references `scripts/compare-fuse-performance-metrics.sh` without explicitly installing it as a dependency. Ensure this script is available in the workflow runner context (appears to rely on checkout).

### Summary

The Go code changes are well-scoped and correct. Extensive test coverage backs every behavioral change. The e2e shell scripts and Python tooling are clean and have their own test suite. No blocking issues.

**Review by mornyx — CHANGES_REQUESTED** [review]
### Review verdict: request changes

I re-reviewed the updated head `b1b554c`. The previous main blockers are improved: clean writable handles now have a read-time refresh path, `OrigSize` is synchronized in the main commit/finalize/rebind paths, and the new focused tests pass locally:

`env GOCACHE=/Users/mornyx/Desktop/repos/github.com/mem9-ai/drive9/.codex-gocache go test ./pkg/fuse -run 'Test(ReadCleanWritableHandleRefreshesSkippedCommittedRevision|ReadOpenHandleSnapshotRetriesTransientRangeRead|OpenWritablePreloadSkipsCleanSiblingReboundToNewRevision)' -count=1 -timeout=5m`

I still see two correctness blockers:

1. `pkg/fuse/dat9fs.go:8594-8600`: the debounced flush path calls `refreshCommittedRevisionForOpenHandles` before publishing the committed size to the inode at line 8617. Since `rebindCleanWriteBufferToRemoteLocked` is called with `committedHandleSizeLocked`, and that helper prefers the inode size, clean sibling handles can adopt the new revision while being rebound to the old size. Please move the inode size update before the sibling refresh, or pass the committed size directly through the refresh/rebind path. This needs a regression covering debounced flush + an already-open clean sibling.

2. `pkg/fuse/dat9fs.go:871-887`: `refreshCleanCommittedRevisionForHandleLocked` returns early for `ShadowReady` / `ShadowSpill` handles. A clean shadow-backed sibling can still be skipped by the non-blocking refresh/cleanup path while locked, then later read from `dirty-shadow` / `shadow-spill` instead of adopting the newer committed revision. If the shadow has already been removed, this can also turn into an avoidable read failure. For clean handles with a newer committed revision, the read-time refresh should clear/rebind stale shadow-backed state instead of returning early, or there should be a separate proof/test that the shadow state is still authoritative for that revision.

Until these two edge cases are fixed, I would keep this PR in request-changes state.

**Comment by qiffang** [comment]
Addressed the latest review blockers in `b26d7c9`.

Fixes:
- Debounced flush now publishes the committed inode size before calling `refreshCommittedRevisionForOpenHandles`, so already-open clean siblings rebind with the committed length instead of the old inode length.
- `refreshCleanCommittedRevisionForHandleLocked` no longer returns early for clean `ShadowReady` / `ShadowSpill` handles when this mount has a newer committed revision. It adopts the new revision, clears stale shadow-backed handle state, and rebinds clean reads to the remote committed view. This covers both removed-shadow cleanup and active stale-shadow state.

Regression coverage:
- `TestDebouncedFlushPublishesSizeBeforeRefreshingCleanSibling`
- `TestReadCleanShadowBackedHandleRefreshesRemovedCommittedShadow`
- `TestReadCleanShadowBackedHandleRefreshesActiveStaleShadow`

Validation:
- `go test ./pkg/fuse -run 'Test(ReadCleanWritableHandleRefreshesSkippedCommittedRevision|ReadCleanShadowBackedHandleRefreshesRemovedCommittedShadow|ReadCleanShadowBackedHandleRefreshesActiveStaleShadow|DebouncedFlushPublishesSizeBeforeRefreshingCleanSibling|OpenWritablePreloadSkipsCleanSiblingReboundToNewRevision)' -count=1 -timeout=5m`
- `go test ./pkg/fuse -count=1 -timeout=5m`
- `go test ./cmd/drive9 ./cmd/drive9/cli ./pkg/fuse -count=1 -timeout=5m`
- `go vet ./pkg/fuse ./cmd/drive9 ./cmd/drive9/cli`
- `git diff --check`
- `git diff --check origin/main...HEAD`
- `PATH="/usr/local/go/bin:$PATH" make lint`

**Review by mornyx — CHANGES_REQUESTED** [review]
### Review verdict: request changes

I re-reviewed `b26d7c9`. The previous fixes are mostly in place, and the existing focused regression tests pass:

`env GOCACHE=/Users/mornyx/Desktop/repos/github.com/mem9-ai/drive9/.codex-gocache go test ./pkg/fuse -run 'Test(ReadCleanWritableHandleRefreshesSkippedCommittedRevision|ReadCleanShadowBackedHandleRefreshesRemovedCommittedShadow|ReadCleanShadowBackedHandleRefreshesActiveStaleShadow|DebouncedFlushPublishesSizeBeforeRefreshingCleanSibling|OpenWritablePreloadSkipsCleanSiblingReboundToNewRevision)' -count=1 -timeout=5m`

Blocking issue:

- `pkg/fuse/dat9fs.go:887` clears `ShadowReady` / `ShadowSpill` on the current clean writable handle when it adopts a newer committed revision, but it leaves the active path shadow in `fs.shadowStore`. A later read-only `Open` can still pin that stale shadow via `PinIfExists` (`pkg/fuse/dat9fs.go:6455`) and `Read` prefers `ReadAtGen` / `ReadAt` from shadow before going remote (`pkg/fuse/dat9fs.go:6810`). I reproduced this by extending `TestReadCleanShadowBackedHandleRefreshesActiveStaleShadow`: after the first handle reads fresh remote data, opening a second `O_RDONLY` handle on the same inode and reading again returns `old active shadow bytes` instead of the committed remote bytes.

Please retire/remove the active stale shadow when a clean shadow-backed handle adopts a newer committed revision, with the usual guard to avoid deleting real pending local state, and add a regression for the second read-only open case.

**Comment by qiffang** [comment]
Addressed the follow-up stale-shadow blocker in `591efa9`.

Fix:
- When a clean `ShadowReady` / `ShadowSpill` handle adopts a newer committed revision, it now retires/removes the active stale shadow before clearing the handle shadow flags and rebinding to the remote committed view.
- The removal is guarded: it will not remove the active shadow if there is pending-index/writeback state, another same-path pending/dirty open handle, or a locked same-path handle that cannot be safely inspected.

Regression:
- Extended `TestReadCleanShadowBackedHandleRefreshesActiveStaleShadow` to reproduce the reported sequence: first clean shadow-backed handle reads fresh committed remote data, then a second `O_RDONLY` open on the same inode reads again and must not pin/read the old active shadow bytes.

Validation:
- `go test ./pkg/fuse -run 'Test(ReadCleanShadowBackedHandleRefreshesActiveStaleShadow|ReadCleanShadowBackedHandleRefreshesRemovedCommittedShadow|DebouncedFlushPublishesSizeBeforeRefreshingCleanSibling|ReadCleanWritableHandleRefreshesSkippedCommittedRevision|OpenWritablePreloadSkipsCleanSiblingReboundToNewRevision)' -count=1 -timeout=5m`
- `go test ./pkg/fuse -count=1 -timeout=5m`
- `go test ./cmd/drive9 ./cmd/drive9/cli ./pkg/fuse -count=1 -timeout=5m`
- `go vet ./pkg/fuse ./cmd/drive9 ./cmd/drive9/cli`
- `git diff --check`
- `git diff --check origin/main...HEAD`
- `PATH="/usr/local/go/bin:$PATH" make lint`

I am also running a fresh all-flags local-e2e on this new head, since this is FUSE read-path correctness code.

**Comment by qiffang** [comment]
Follow-up validation on latest head `591efa9` is complete.

Fresh GitHub checks:
- `ci` run `27121739766`: success
- pull_request `local-e2e smoke` run `27121739805`: success
- all-flags workflow_dispatch run `27121776936`: success
  - release gate: success
  - performance artifact upload: success
  - Drive9 archive: success
  - FUSE concurrency stress: success
  - POSIX/fsx gate: success
  - final enforce: success
- CodeRabbit status: success

Artifact verification from `27121776936`:
- `actual-mounted-manifest.json == expected-manifest.json`
- `actual-remote-manifest.json == expected-manifest.json`
- `reader-errors.log` is empty
- SQLite rollback read: `payload_verified_rows=256`, `integrity_check=ok`, `journal_mode=delete`
- SQLite WAL read: `payload_verified_rows=256`, `integrity_check=ok`, `journal_mode=wal`
- WAL checkpoint: `checkpoint_busy=0`, `integrity_check=ok`

@mornyx please re-review latest head `591efa9`; the active stale shadow/read-only reopen blocker should now be covered by `TestReadCleanShadowBackedHandleRefreshesActiveStaleShadow`.

**Review by mornyx — APPROVED** [review]
### Review verdict: approve

Re-reviewed latest head `591efa9`.

The previous stale active shadow blocker is addressed. Clean shadow-backed handles now retire the active path shadow before clearing shadow state and rebinding, with guards for pending index/writeback state and other open handles that still have local state. The regression also covers the read-only reopen sequence that previously reproduced stale `ShadowStore` reads.

Focused validation passed:
`env GOCACHE=/Users/mornyx/Desktop/repos/github.com/mem9-ai/drive9/.codex-gocache go test ./pkg/fuse -run "Test(ReadCleanShadowBackedHandleRefreshesActiveStaleShadow|ReadCleanShadowBackedHandleRefreshesRemovedCommittedShadow|DebouncedFlushPublishesSizeBeforeRefreshingCleanSibling|ReadCleanWritableHandleRefreshesSkippedCommittedRevision|OpenWritablePreloadSkipsCleanSiblingReboundToNewRevision)" -count=1 -timeout=5m`

I did not find remaining blocking issues in this pass.
